### PR TITLE
(Outdated -- use #175) build subnetworks

### PR DESCRIPTION
**Pull request by awlostowski-noaa**
A new utility to build subnetworks of common order from larger networks.

### CODE REVIEW

**Comment by awlostowski-noaa** [comment]
# suggested improvement
Currently, subnetwork creation is constrained by a user-appointed maximum subnetwork depth. Subnetworks are created via a breadth-first-search. The search begins from a source node and stops once the maximum depth is reached OR once there are no more nodes to reach.

**Should we change slightly such that the search continues until either 1) a threshold number of segments are in the subnetwork or 2) there are no more nodes to reach?** The issue with the code as it stands now, is that the depth of the subnetwork is not immediately related to the size (number of segments) in the subnetwork.

File: src/python_framework_v02/nhd_network.py [hunk 1]
```diff
@@ -417,3 +417,85 @@ def replace_waterbodies_connections(connections, waterbodies):
...
+                    reachable.add(x)
+                    reachable_depth.add(y)
+
+                    if len(rconn.get(x, ())) > 1:
+                        us_depth = y+1
+                    else:
+                        us_depth = y
+
+                    if us_depth <= max_depth:
+                        D.extend([us_depth]*len(rconn.get(x, ())))
```
Comment (jameshalgren) [inline src/python_framework_v02/nhd_network.py:471]: Need additional criteria for hitting a minimum number of segments aggregated into the group.

**Review by jameshalgren — DISMISSED** [review]

**Comment by jameshalgren** [comment]
Replaced by #175

File: src/python_framework_v02/nhd_network.py (hunk 1 above)
Comment (jameshalgren) [reply src/python_framework_v02/nhd_network.py]: I think we have this functionality as it now stands in PR #175 (built from this PR code) So I'll resolve this conversation.

**Comment by jameshalgren** [comment]
As noted, closing to use #175 which contains this code rebased to the current master + additional capability for using the grouped/clustered subnetworks in the actual routing function.

**Pull request closed without merging**